incubator/hnc/internal/reconcilers/object.go

```diff
@@ -540,6 +540,25 @@ func (r *ObjectReconciler) writeObject(ctx context.Context, log logr.Logger, ins
 	if exist {
 		log.Info("Updating")
 		err = r.Update(ctx, inst)
+		// RoleBindings can't have their Roles changed after they're created. 
+		// If an RB was quickly delete and re-created in an ancestor namespace 
+		// - fast enough that by the time that HNC notices, the new RB exists - 
+		// HNC could see it as an update (not a delete + create) and attempt to 
+		// update the RBs in all descendant namespaces, and this will fail. 
+		// In order to handle this case, we try to delete and re-create the 
+		// rolebinding here
```

Comment:
This is a good comment, but it only talks about RBs while the code is general. We should explain why that is.

Reply:
Also you talk about "quickly" deleting and re-creating, but this probably also comes up if there's a change while HNC isn't running (which is what you actually test). Maybe add that to the comment as well.

Reply:
Done. Added a link to why the RB is immutable